src/main/java/Duke.java

```diff
@@ -1,10 +1,124 @@
-    public static void main(String[] args) {
+    public static void main(String[] args) throws IOException {
+
         String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        String nextLine = br.readLine().strip().replaceAll(" +", " ");
+
+        ArrayList<Task> list = new ArrayList<Task>();
+
+        while (!nextLine.equals("bye")) {
+            System.out.println("__________________________________________________");
+            String firstWord = nextLine.split(" ")[0];
+            if (firstWord.equals("mark")) {
+                try {
+                    int indexToMark = Integer.parseInt(nextLine.substring(5)) - 1;
+                    list.set(indexToMark, list.get(indexToMark).mark());
+                    System.out.println("Nice! I've marked this task as done:\n  " 
+                            + list.get(indexToMark));
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Please choose a valid task! (Your list has " 
+                            + list.size() + " tasks)");
+                } catch (NumberFormatException e) {
+                    System.out.println("Please input an integer!");
+                }
+            } else if (firstWord.equals("unmark")) {
+                try {
+                    int indexToUnmark = Integer.parseInt(nextLine.substring(7)) - 1;
+                    list.set(indexToUnmark, list.get(indexToUnmark).unmark());
+                    System.out.println("OK, I've marked this task as not done yet:\n  " 
+                            + list.get(indexToUnmark));
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Please choose a valid task! (Your list has " 
+                            + list.size() + " tasks)");
+                } catch (NumberFormatException e) {
+                    System.out.println("Please input an integer!");
+                }
+            } else if (firstWord.equals("list")) {
+                if (nextLine.split(" ").length > 1) {
+                    System.out.println("Uh-oh! There should not be any words after 'list'.");
+                } else if (list.isEmpty()) {
+                    System.out.println("Uh-oh! List is empty!");
+                } else {
+                    for (int i = 0; i < list.size(); i++) {
+                        System.out.println((i + 1) + "." + list.get(i));
+                    }
+                }
+            } else if (firstWord.equals("delete")) {
```

Comment:
Perhaps it would be good if you can create more classes e.g. Parser with specific purposes. This way, you might be able to avoid deep nesting